cogs5e/lookup.py

```diff
@@ -451,29 +455,33 @@ async def monimage(self, ctx, *, name: str):
         await ctx.send(embed=embed)
 
     @commands.command()
-    async def token(self, ctx, name=None, *args):
+    async def token(self, ctx, *, name=""):
         """
         Shows a monster or your character's token.
         __Valid Arguments__
         -border <plain|none (player token only)> - Overrides the token border.
         """
-        if name is None or name.startswith("-"):
+        if not name or name.startswith("-"):
             token_cmd = self.bot.get_command("playertoken")
             if token_cmd is None:
                 return await ctx.send("Error: SheetManager cog not loaded.")
-            if name:
-                args = (name, *args)
-            return await ctx.invoke(token_cmd, *args)
+            return await ctx.invoke(token_cmd, name)
 
         # select monster
+        token_args = argparse(name)
+        hide_name = token_args.get("h", False, bool)
+        if not name.find("-") == -1:  # remove the args from the name
+            name = name[: name.find("-")].rstrip()
```

Comment:
Imo we shouldn't add this logic and stick to the old `async def token(self, ctx, name=None, *args)` signature instead - this feels hacky